### PR TITLE
preauthorization tests 

### PR DESCRIPTION
**Pull request by mchalski**
https://tracker.mender.io/browse/MEN-1640

@mendersoftware/rndity @maciejmrowiec  @GregorioDiStefano 

NOTE: integration tests have been failing consistently on unrelated test cases - I'll keep restarting them   until maybe finally we'll get a green build. either way - the tests in this PR are green locally on my machine.

### CODE REVIEW

File: tests/tests/test_preauth.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+        new_tenant_client("tenant-container", token)
+
+
+class Client:
+    """Wraps various actions on the client, performed via SSH (inside fabric.execute())."""
+
+    ID_HELPER = '/usr/share/mender/identity/mender-device-identity'
+    PRIV_KEY = '/data/mender/mender-agent.pem'
+    KEYGEN_TIMEOUT = 120
```
Comment (GregorioDiStefano) [inline tests/tests/test_preauth.py]: Increase the timeout

File: tests/tests/test_preauth.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+        assert dev_preauth['device_identity'] == preauth_iddata_str
+        assert dev_preauth['key'] == preauth_key
+
+        # make one of the existing devices the preauthorized device
+        # by substituting id data and restarting
+        res = execute(Client.substitute_id_data, preauth_iddata, hosts=client)
+        res = execute(Client.restart, hosts=client)
+
+        # verify api results - after some time the device should be 'accepted'
+        time.sleep(30)
```
Comment (GregorioDiStefano) [inline tests/tests/test_preauth.py]: Increase this timeout

File: tests/tests/test_preauth.py (hunk 1 above)
Comment (mchalski) [reply tests/tests/test_preauth.py]: what's your recommendation?

File: tests/tests/test_preauth.py (hunk 2 above)
Comment (mchalski) [reply tests/tests/test_preauth.py]: what's your recommendation?

Comment (mchalski) [reply tests/tests/test_preauth.py]: btw this one should be instantaneous, I just restart the mender binary inside quemu (not the client itself)

Comment (mchalski) [reply tests/tests/test_preauth.py]: also, the tests are green. if we don't have good ideas for a precise number I'd leave it well alone for the time being.

**Comment by GregorioDiStefano** [comment]
as expected, the 120 seconds timeout was exceeded in the second test run since this was merged, i increased it here:  https://github.com/mendersoftware/integration/pull/437/files

**Pull request merged**